pkg/policy/selectorcache.go

```diff
@@ -136,6 +137,26 @@ func (sc *SelectorCache) GetModel() models.SelectorCache {
 	return selCacheMdl
 }
 
+func (sc *SelectorCache) Stats() selectorStats {
+	result := newSelectorStats()
+
+	sc.mutex.RLock()
+	defer sc.mutex.RUnlock()
```

Comment:
I don't have a good sense for the impact of this. It's just the readlock, but how often does Prometheus call the `Collect()` function on a `Collector`? cc @cilium/metrics

Reply:
It occurs on each scrape, so it depends on your Prometheus scrape interval.